jax/api_util.py

```diff
@@ -74,16 +75,78 @@ def flatten_fun_nokwargs2(in_tree, *args_flat):
   aux_flat, aux_tree = tree_flatten(aux)
   yield (ans_flat, aux_flat), (ans_tree, aux_tree)
 
-def argnums_partial(f, dyn_argnums, args):
+
+def argnums_partial(f, dyn_argnums, args, raise_error=False):
+  """Keeps only dynamic arguments for the arguments of f.
+
+  Args:
+    f: The function being wrapped.
+    dync_arguments: The indices of the dynamic arguments (i.e. non-static
+      positial arguments).
+    args: The positional arguments.
+
+  Returns:
+    A tuple of:
+      The function wrapped into a partial with the non dynamic arguments
+        wrapped.
+      The tuple of the dynamic arguments.
+  """
   if isinstance(dyn_argnums, int):
     dyn_argnums = (dyn_argnums,)
-  else:
+  elif not isinstance(dyn_argnums, tuple):
     dyn_argnums = tuple(dyn_argnums)
-  fixed_args = tuple([unit if i in dyn_argnums else wrap_hashably(arg)
-                      for i, arg in enumerate(args)])
+  fixed_args = tuple(
+      unit if i in dyn_argnums else wrap_hashably(arg)
+      for i, arg in enumerate(args))
   dyn_args = tuple(args[i] for i in dyn_argnums)
   return _argnums_partial(f, dyn_argnums, fixed_args), dyn_args
 
+
+def argnums_partial_except(f,
+                           static_argnums: Tuple[int, ...],
+                           args,
+                           raise_error=False):
```

Comment:
Can we remove the `raise_error` argument? I don't think it's used by any callers, and having options on a function is kind of an anti-pattern. Instead, we can just have the version of the function body that prints a warning for now, and in the future edit it to raise an error. (If we want to make it switchable later, we could have a flag rather than an argument to an internal-only function.)

Reply:
Done.